src/main/java/mezz/jei/plugins/vanilla/anvil/AnvilRecipeWrapper.java

```diff
@@ -0,0 +1,77 @@
+package mezz.jei.plugins.vanilla.anvil;
+
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.BlankRecipeWrapper;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.item.ItemStack;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class AnvilRecipeWrapper extends BlankRecipeWrapper {
+	private final List<List<ItemStack>> inputs;
+	private final List<List<ItemStack>> output;
+	private final int cost;
+
+	public AnvilRecipeWrapper(ItemStack leftInput, ItemStack rightInput, ItemStack output) {
+		this(leftInput, rightInput, output, -1);
+	}
+
+	public AnvilRecipeWrapper(ItemStack leftInput, ItemStack rightInput, ItemStack output, int levelsCost) {
+		this(leftInput, Collections.singletonList(rightInput), Collections.singletonList(output), levelsCost);
+	}
+
+	public AnvilRecipeWrapper(ItemStack leftInput, List<ItemStack> rightInputs, List<ItemStack> outputs) {
+		this(leftInput, rightInputs, outputs, -1);
+	}
+
+	@SuppressWarnings("unchecked")
```

Comment:
I don't think it's possible to use `Arrays.asList` with genetics, you're right.
Try creating a new arraylist and then adding the inputs to it after creation.